forte/mrdsrg-spin-integrated/dsrg_mrpt2_gradient.cc

```diff
@@ -0,0 +1,3996 @@
+        Tau2["UIXA"] += 0.25 * DelGam1["XU"] * Sigma2["IA"];
+        Tau2["uIxA"] += 0.25 * DelGam1["xu"] * Sigma2["IA"];
+
+        Tau2["uiax"] -= 0.25 * DelGam1["xu"] * Sigma2["ia"];
+        Tau2["UIAX"] -= 0.25 * DelGam1["XU"] * Sigma2["IA"];
+    }
+    // <[F, T2]>
+    if (X5_TERM) {
+        Tau2["uvey"] += 0.125 * F_["ex"] * Lambda2_["xyuv"];
+        Tau2["UVEY"] += 0.125 * F_["EX"] * Lambda2_["XYUV"];
+        Tau2["uVeY"] += 0.125 * F_["ex"] * Lambda2_["xYuV"];
+
+        Tau2["vuye"] += 0.125 * F_["ex"] * Lambda2_["yxvu"];
+        Tau2["VUYE"] += 0.125 * F_["EX"] * Lambda2_["YXVU"];
+        Tau2["vUyE"] += 0.125 * F_["EX"] * Lambda2_["yXvU"];
+
+        Tau2["uvye"] += 0.125 * F_["ex"] * Lambda2_["yxuv"];
+        Tau2["UVYE"] += 0.125 * F_["EX"] * Lambda2_["YXUV"];
+        Tau2["uVyE"] += 0.125 * F_["EX"] * Lambda2_["yXuV"];
+
+        Tau2["vuey"] += 0.125 * F_["ex"] * Lambda2_["xyvu"];
+        Tau2["VUEY"] += 0.125 * F_["EX"] * Lambda2_["XYVU"];
+        Tau2["vUeY"] += 0.125 * F_["ex"] * Lambda2_["xYvU"];
+    
+        Tau2["umxy"] -= 0.125 * F_["vm"] * Lambda2_["xyuv"];
+        Tau2["UMXY"] -= 0.125 * F_["VM"] * Lambda2_["XYUV"];
+        Tau2["uMxY"] -= 0.125 * F_["VM"] * Lambda2_["xYuV"];
+
+        Tau2["muyx"] -= 0.125 * F_["vm"] * Lambda2_["yxvu"];
+        Tau2["MUYX"] -= 0.125 * F_["VM"] * Lambda2_["YXVU"];
+        Tau2["mUyX"] -= 0.125 * F_["vm"] * Lambda2_["yXvU"];
+
+        Tau2["umyx"] -= 0.125 * F_["vm"] * Lambda2_["yxuv"];
+        Tau2["UMYX"] -= 0.125 * F_["VM"] * Lambda2_["YXUV"];
+        Tau2["uMyX"] -= 0.125 * F_["VM"] * Lambda2_["yXuV"];
+
+        Tau2["muxy"] -= 0.125 * F_["vm"] * Lambda2_["xyvu"];
+        Tau2["MUXY"] -= 0.125 * F_["VM"] * Lambda2_["XYVU"];
+        Tau2["mUxY"] -= 0.125 * F_["vm"] * Lambda2_["xYvU"];
+    }
+
+    if (CORRELATION_TERM) {
+        Tau2["iuax"] += 0.25 * DelGam1["xu"] * Xi3["ia"];
+        Tau2["IUAX"] += 0.25 * DelGam1["XU"] * Xi3["IA"];
+        Tau2["iUaX"] += 0.25 * DelGam1["XU"] * Xi3["ia"];
+
+        Tau2["iuxa"] -= 0.25 * DelGam1["xu"] * Xi3["ia"];
+        Tau2["IUXA"] -= 0.25 * DelGam1["XU"] * Xi3["IA"];
+
+        Tau2["uixa"] += 0.25 * DelGam1["xu"] * Xi3["ia"];
+        Tau2["UIXA"] += 0.25 * DelGam1["XU"] * Xi3["IA"];
+        Tau2["uIxA"] += 0.25 * DelGam1["xu"] * Xi3["IA"];
+
+        Tau2["uiax"] -= 0.25 * DelGam1["xu"] * Xi3["ia"];
+        Tau2["UIAX"] -= 0.25 * DelGam1["XU"] * Xi3["IA"];
+    }
+
+    // NOTICE: remove the internal parts based on the DSRG theories
+    Tau2.block("aaaa").zero();
+    Tau2.block("aAaA").zero();
+    Tau2.block("AAAA").zero();
+
+    // Tau * [1 - e^(-s * Delta^2)]
+    Tau1["ijab"] = Tau2["ijab"] * Eeps2_m1["ijab"];
+    Tau1["IJAB"] = Tau2["IJAB"] * Eeps2_m1["IJAB"];
+    Tau1["iJaB"] = Tau2["iJaB"] * Eeps2_m1["iJaB"];
+
+    outfile->Printf("Done");
+}
+
+void DSRG_MRPT2::set_sigma() {
+    outfile->Printf("\n    Initializing Sigma .............................. ");
```

Comment:
Please be more specific on the printing, like `Initializing multipliers for ...`.

Reply:
fixed